frontend/public/components/factory/details.tsx

```diff
@@ -44,7 +44,7 @@ export const DetailsPage = withFallback<DetailsPageProps>((props) => {
         }
         resourceKeys={resourceKeys}
         customData={props.customData}
-        badge={props.badge}
+        badge={props.badge || getBadgeFromType(props.kindObj.badge)}
```

Comment:
Allowed for an override... if this is undesirable as all models should use badges and we have no other uses for the override, I can remove it.